website/docs/plugin/framework/migrating/attributes-blocks/types.mdx

```diff
@@ -0,0 +1,202 @@
+
+- In the Framework, an `Attribute` struct has either the `Type` or `Attributes` field defined. The `Type` field is set to
+a primitive type such as an integer or string, and you use the `Attributes` field for `NestedAttributes`. Refer to
+[Provider Schema](/plugin/framework/migrating/providers#example-1) for an example of using a single
+nested attribute. Nested attributes are also described in more detail on the
+[Schemas](https://www.terraform.io/plugin/framework/schemas#attributes-1) page in the Framework documentation.
+
+## Example
+
+The following examples show how to migrate portions of the
+[http](https://github.com/hashicorp/terraform-provider-http) provider.
+
+For a complete example, clone the
+`terraform-provider-http` repository and compare the `data_source.go` file in
+[v2.2.0](https://github.com/hashicorp/terraform-provider-http/blob/v2.2.0/internal/provider/data_source.go)
+and the `data_source_http.go` file in
+[v3.0.1](https://github.com/hashicorp/terraform-provider-http/blob/v3.0.1/internal/provider/data_source_http.go).
+
+### SDKv2
+
+The following example from the `data_source.go` file shows the implementation of the type field of the `url` attribute
+for the `http` data source with SDKv2.
+
+```go
+func dataSource() *schema.Resource {
+    return &schema.Resource{
+        Schema: map[string]*schema.Schema{
+            "url": {
+                Type:        schema.TypeString,
+                /* ... */
+            },
+            /* ... */
+```
+
+### Framework
+
+The following example from the `data_source_http.go` file shows how the type of the `url` attribute for the `http` data
+source is defined with the Framework after the migration.
+
+```go
+func (d *httpDataSourceType) GetSchema(context.Context) (tfsdk.Schema, diag.Diagnostics) {
+    return tfsdk.Schema{
+        Attributes: map[string]tfsdk.Attribute{
+            "url": {
+                Type:        types.StringType,
+                /* ... */
+            },
+            /* ... */
+```
+
+The following examples show how to migrate portions of the
+[http](https://github.com/hashicorp/terraform-provider-tls) provider.
+
+For a complete example, clone the
+`terraform-provider-tls` repository and compare the `common_cert.go` file in
+[v3.4.0](https://github.com/hashicorp/terraform-provider-tls/blob/v3.4.0/internal/provider/common_cert.go)
+and the `resource_cert_request.go` file in
+[v4.0.1](https://github.com/hashicorp/terraform-provider-tls/blob/v4.0.1/internal/provider/resource_cert_request.go).
+
+### SDKv2
+
+The following example from the `common_cert.go` file shows the implementation of the type field of the `dns_names`
+attribute with SDKv2.
+
+```go
+func resourceCertRequest() *schema.Resource {
+    return &schema.Resource{
+        Schema: map[string]*schema.Schema{
+            "dns_names": {
+                Type:    schema.TypeList,
+				Elem:    &schema.Schema{
```

Comment:
`Elem` should be indented at the same level as `Type` on the line above, right?

Reply:
Fixed